ユーザー管理機能の実装

mayumishibata-div opened the pull request:
# WHAT
ユーザーの新規登録、ログイン、ログアウトができるように実装
# WHY
ユーザーに対してアカウントを発行しそのアカウントでユーザーの情報を管理するため
■以下の手順で実装
　手順1 deviseを導入する
　手順2 新規登録機能を実装する（バリデーション、エラーメッセージの設定）
　手順3 ログイン、ログアウト機能を実装する
　手順4 モデルの単体テストを実装する

＜確認画像＞
◉必要な情報を適切に入力して「会員登録」ボタンを押すと、ユーザーの新規登録ができる動画
https://gyazo.com/688068899dc91faea2c9e7a60cd00591

◉入力に問題のある状態で「会員登録」ボタンが押された場合、ユーザーの新規登録はできず、エラーメッセージが出力される動画
https://gyazo.com/b91f4853b6755095066a1fc438486b5b

◉必要な情報を適切に入力して「ログイン」ボタンを押すと、ユーザーのログインができる動画
https://gyazo.com/084cb47ee54b32b4705e2cc81774c606

◉入力に問題のある状態で「ログイン」ボタンが押された場合、ユーザーのログインはできず、エラーメッセージが出力される動画
https://gyazo.com/eb5f7359289ad87cd5771310e2698c59

◉ログアウトに成功する動画
https://gyazo.com/dada8be82d0648d3fc56582339737720

◉テストに成功する画像
<img width="1008" alt="スクリーンショット 2025-02-21 13 23 38" src="https://github.com/user-attachments/assets/45d4d95b-034e-4970-9740-26d0b9ee72b9" />

mayumishibata-div committed:
deviseのGemをインストール

mayumishibata-div committed:
ユーザーを管理するUserモデルを作成

mayumishibata-div committed:
必要なカラムを記述し、マイグレーションを実行

mayumishibata-div committed:
deviseのビューを配置

mayumishibata-div committed:
ログイン画面に遷移するように実装

mayumishibata-div committed:
カラム情報をDBに保存

mayumishibata-div committed:
ログイン画面に遷移する実装を一旦コメントアウト

mayumishibata-div committed:
ユーザー新規登録のための実装

mayumishibata-div committed:
ログインの有無で表示を帰る実装

mayumishibata-div committed:
ログアウトの実装

mayumishibata-div committed:
ログイン機能の実装

mayumishibata-div committed:
バリデーションの設定

mayumishibata-div committed:
全角かな/カナ漢字の区別のバリデーション

mayumishibata-div committed:
passwordの英数混合の判断

mayumishibata-div committed:
エラーメッセージの表示

mayumishibata-div committed:
パスワードが6文字以上のバリデーションを設定

mayumishibata-div committed:
テストコードの準備

mayumishibata-div committed:
テストコードの実装

mayumishibata-div committed:
Update README.md

TakashiKozuma commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -0,0 +1,108 @@
+
+  <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
+  <%= render 'shared/error_messages', model: f.object %>
+  <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
```rb
<%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
3,5,13,15行目のコメントアウトは実装案内なので削除しておきましょう。
実装案内は完成したアプリを人に見せる際に残しておくの望ましくないためです。
他にも「〜しましょう」のような実装案内があれば、実装時に削除するようにしていきましょう。

mayumishibata-div replied:
実装案内削除

TakashiKozuma commented on app/views/shared/_header.html.erb:
```diff
@@ -16,13 +16,15 @@
+    <% else %>
+      <li><%= link_to 'ログイン', new_user_session_path, class: "login" %></li>
+      <li><%= link_to '新規登録', new_user_registration_path, class: "sign-up" %></li>
       <%# //deviseを導入できたら、ログインの有無で表示が変わるように分岐しましょう%>
```
```Rb
<%# //deviseを導入できたら、ログインの有無で表示が変わるように分岐しましょう%>
```
18,26行目も実装案内なので、削除しましょう。

mayumishibata-div replied:
実装案内削除

TakashiKozuma commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,109 @@
+        @user.valid?
+        expect(@user.errors.full_messages).to include("Birthday can't be blank")
+      end
+    end
```
以下の内容もテストしましょう！
- passwordに全角文字が含まれていると登録できない。
- お名前(全角)は、全角（漢字・ひらがな・カタカナ）での入力が必須であること
- お名前カナ(全角)は、全角（カタカナ）での入力が必須であること

mayumishibata-div replied:
テストコードの追加
・passwordに全角文字が含まれていると登録できない。
・お名前(全角)は、全角（漢字・ひらがな・カタカナ）での入力が必須であること
・お名前カナ(全角)は、全角（カタカナ）での入力が必須であること

TakashiKozuma reviewed (commented):
レビュー依頼の内容を確認致しました。
コメントを致しましたので、ご確認お願いします。

修正後はコミット・プッシュのし忘れがないかご確認の上、再度フォームからレビュー依頼をお願い致します。

mayumishibata-div committed:
実装案内の削除

mayumishibata-div committed:
テストコードの追加

passwordに全角文字が含まれていると登録できない。
名前(全角)は、全角（漢字・ひらがな・カタカナ）での入力が必須であること
名前が半角英数字、半角カタカナでは登録できないこと
名前カナ(全角)は、全角（カタカナ）での入力が必須であること

mayumishibata-div commented:
<img width="642" alt="スクリーンショット 2025-02-23 9 36 44" src="https://github.com/user-attachments/assets/ab09c05c-ee68-4ad4-b70f-8f59613d2be4" />

TakashiKozuma commented:
LGTMです！実装お疲れさまでした。

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。
